pkg/internal/apis/acme/install/pruning_test.go

```diff
@@ -17,20 +17,59 @@ limitations under the License.
 package install
 
 import (
-	"fmt"
+	"bytes"
+	"io/ioutil"
 	"testing"
 
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	apiextensionsinstall "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/install"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	jsonserializer "k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/apimachinery/pkg/runtime/serializer/versioning"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+
 	crdfuzz "github.com/munnerz/crd-schema-fuzz"
 
 	"github.com/jetstack/cert-manager/pkg/api"
 	acmefuzzer "github.com/jetstack/cert-manager/pkg/internal/apis/acme/fuzzer"
 )
 
 func TestPruneTypes(t *testing.T) {
-	crdfuzz.SchemaFuzzTestForCRDWithPath(t, api.Scheme, crdPath("orders"), acmefuzzer.Funcs)
-	crdfuzz.SchemaFuzzTestForCRDWithPath(t, api.Scheme, crdPath("challenges"), acmefuzzer.Funcs)
+	crdfuzz.SchemaFuzzTestForInternalCRD(t, api.Scheme, getCRD(t, "orders.acme.cert-manager.io"), acmefuzzer.Funcs)
+	crdfuzz.SchemaFuzzTestForInternalCRD(t, api.Scheme, getCRD(t, "challenges.acme.cert-manager.io"), acmefuzzer.Funcs)
 }
 
-func crdPath(s string) string {
-	return fmt.Sprintf("../../../../../deploy/charts/cert-manager/crds/%s.yaml", s)
+func getCRD(t *testing.T, resourceName string) *apiextensions.CustomResourceDefinition {
+	internalScheme := runtime.NewScheme()
+	utilruntime.Must(metav1.AddMetaToScheme(internalScheme))
+	apiextensionsinstall.Install(internalScheme)
+	serializer := jsonserializer.NewSerializerWithOptions(jsonserializer.DefaultMetaFactory, internalScheme, internalScheme, jsonserializer.SerializerOptions{
+		Yaml: true,
+	})
+	convertor := runtime.UnsafeObjectConvertor(internalScheme)
+	codec := versioning.NewCodec(serializer, serializer, convertor, internalScheme, internalScheme, internalScheme, runtime.InternalGroupVersioner, runtime.InternalGroupVersioner, internalScheme.Name())
+
+	data, err := ioutil.ReadFile("../../../../../deploy/manifests/00-crds.yaml")
+	if err != nil {
+		t.Fatalf("Failed to read CRD input file: %v", err)
+		return nil
+	}
+
+	individualCRDs := bytes.Split(data, []byte("---"))
+	for _, crdData := range individualCRDs {
```

Comment:
There is {some way} to make the decoder automatically understand multiple CRD objects by passing it an `io.Reader`, but I'm not too sure how to do it within the `versioning.NewCodec` function and this works just fine (until one day we include `---` in our OpenAPI definition, but at least those tests will just fail 😄)